pkg/openfeature/event_executor_test.go

```diff
@@ -0,0 +1,529 @@
+package openfeature
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/open-feature/go-sdk/pkg/openfeature/internal"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sync/errgroup"
+	"reflect"
+	"testing"
+	"time"
+)
+
+var logger logr.Logger
+
+func init() {
+	logger = logr.New(internal.Logger{})
+}
+
+// Requirement 5.1.1 The provider MAY define a mechanism for signaling the occurrence of one of a set of events,
+// including PROVIDER_READY, PROVIDER_ERROR, PROVIDER_CONFIGURATION_CHANGED and PROVIDER_STALE,
+// with a provider event details payload.
+func TestEventHandler_RegisterUnregisterEventProvider(t *testing.T) {
+
+	t.Run("Ignored non-eventing providers", func(t *testing.T) {
+		handler := newEventHandler(logger)
+		handler.registerEventingProvider(NoopProvider{})
+
+		if len(handler.providerShutdownHook) != 0 {
+			t.Errorf("implementation should ignore non eventing provider")
+		}
+
+		err := handler.unregisterEventingProvider(NoopProvider{})
+		if err != nil {
+			t.Error(err)
+		}
+	})
+
+	t.Run("Accepts addition and removal of eventing providers", func(t *testing.T) {
+		eventingImpl := &ProviderEventing{}
+
+		eventingProvider := struct {
+			FeatureProvider
+			EventHandler
+		}{
+			NoopProvider{},
+			eventingImpl,
+		}
+
+		handler := newEventHandler(logger)
```

Comment:
`hander` and `newEventHandler` are both confusing to me here. This is not a handler, and the function doesn't seem to return a `handler`... but an `eventExecutor`, which is very different conceptually from a handler.

Reply:
This is something missed from renaming [1] . will fix in the next commits 

[1] - https://github.com/open-feature/go-sdk/pull/188/files#diff-5088e3f82f28056163763bfb774c1b8e91bb585580b4b1add90b133445b8bae6R26-R33

Reply:
Rename completed